lib/ansible/modules/cloud/podman/podman_container.py

```diff
@@ -0,0 +1,1213 @@
+            for pub in params['publish']:
+                cmd += ['--publish', pub]
+        if params.get('publish_all') is not None:
+            cmd += ['--publish-all=' + str(params['publish_all'])]
+        if params.get('read_only') is not None:
+            cmd += ['--read-only=' + str(params['read_only'])]
+        if params.get('read_only_tmpfs') is not None:
+            cmd += ['--read-only-tmpfs=' + str(params['read_only_tmpfs'])]
+        if params.get('restart_policy'):
+            cmd += ['--restart=' + params['restart_policy']]
+        if params.get('rm') is not None:
+            cmd += ['--rm=' + str(params['rm'])]
+        if params.get('rootfs') is not None and params.get('rootfs'):
+            cmd += ['--rootfs']
+        if params.get('security_opt'):
+            cmd += ['--security-opt', params['security_opt']]
+        if params.get('shm_size'):
+            cmd += ['--shm-size', params['shm_size']]
+        if params.get('sig_proxy') is not None:
+            cmd += ['--sig-proxy=' + str(params['sig_proxy'])]
+        if params.get('stop_signal'):
+            cmd += ['--stop-signal', params['stop_signal']]
+        if params.get('stop_timeout'):
+            cmd += ['--stop-timeout', str(params['stop_timeout'])]
+        if params.get('subgidname'):
+            cmd += ['--subgidname', params['subgidname']]
+        if params.get('subuidname'):
+            cmd += ['--subuidname', params['subuidname']]
+        if params.get('sysctl'):
+            cmd += ['--sysctl', params['sysctl']]
+        if params.get('systemd') is not None:
+            cmd += ['--systemd=' + str(params['systemd'])]
+        if params.get('tmpfs'):
+            cmd += ['--tmpfs', params['tmpfs']]
+        if params.get('tty') is not None:
+            cmd += ['--tty=' + str(params['tty'])]
+        if params.get('uidmap'):
+            cmd += ['--uidmap', params['uidmap']]
+        if params.get('ulimit'):
+            cmd += ['--ulimit', params['ulimit']]
+        if params.get('user'):
+            cmd += ['--user', params['user']]
+        if params.get('userns'):
+            cmd += ['--userns', params['userns']]
+        if params.get('uts'):
+            cmd += ['--uts', params['uts']]
+        if params.get('volume'):
+            for vol in params['volume']:
+                cmd += ['--volume', vol]
+        if params.get('volumes_from'):
+            for vol in params['volumes_from']:
+                cmd += ['--volumes-from', vol]
+        if params.get('workdir'):
+            cmd += ['--workdir', params['workdir']]
+        # Add your own args for podman command
+        if params.get('cmd_args'):
+            cmd += params['cmd_args']
+        cmd.append(params['image'])
+        if params.get('command'):
+            cmd.append(params['command'])
+        return cmd
+
+
+def ensure_image_exists(module, image):
+    rc, out, err = run_podman_command(module, args=['image', 'exists', image], ignore_errors=True)
+    if rc != 0:
```

Comment:
plz, reverse such checks into guard expressions to make the code flatter.

<details>
<summary>Details</summary>

![](https://pbs.twimg.com/media/D5tgy4cWsAMk3kf.png)
![](https://i.stack.imgur.com/z9Gil.png)

</details>

Reply:
changed this function to save calls